Issue2948 MotorDrive package

mingzhe37 opened the pull request:
Hi Dr. Hu @JayHuLBL, we would like to start the model review process for merging the MotorDrive package into the master branch through this pull request. In this pull request, we include squirrel-type induction motor models, pump/heat pump/chiller models with a mechanical interface, and motor coupled pump/heat pump/chiller models. Besides, two interface models, PartialOnePort and PartialBaseOnePort, are added to be used in motor related models, as listed below.

- Buildings.Electrical.AC.ThreePhasesBalanced.Loads.MotorDrive:
　**InductionMotors**:
　　SquirrelCage
　　SquirrelCageDrive
　　**Examples**:
　　　SquirrelCage
　　　SquirrelCageDrive
　　**BaseClasses**:
　　　MotorMachineInterface
　　　**Validation**:
　　　　MotorMachineInterface 
　**ThermoFluid**:
　　Chiller
　　HeatPump
　　Pump
　　**Examples**:
　　　Chiller
　　　HeatPump
　　　Pump
　**Coupled**:
　　Chiller
　　HeatPump
　　Pump
　　**Examples**:
　　　Chiller
　　　HeatPump
　　　Pump
- Buildings.Electrical.Interfaces
　PartialOnePort 
　PartialBaseOnePort 

Unit test for this package has passed with Dymola 2022x. This PR is related to issue https://github.com/lbl-srg/modelica-buildings/issues/2948. Please let us know if you have any questions. Thanks you very much!

JayHuLBL commented on Buildings/Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/InductionMotors/BaseClasses/MotorMachineInterface.mo:
```diff
@@ -0,0 +1,106 @@
+<p align=\"center\"><img src=\"modelica://Buildings/Resources/Images/Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/InductionMotors/induction_motor_equivalent_circuit.png\"/> </p>
+<p>The electromagnetic torque generated by motor is calculated as follows. </p>
+<p align=\"center\"><img src=\"modelica://Buildings/Resources/Images/Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/InductionMotors/torque_euqation.png\"/> </p>
+Where, <br>
```
missing `<p>`

mingzhe37 replied:
Thanks for finding this issue, missing characters in all files have been resolved.

JayHuLBL commented on Buildings/Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/InductionMotors/BaseClasses/MotorMachineInterface.mo:
```diff
@@ -0,0 +1,106 @@
+<p align=\"center\"><img src=\"modelica://Buildings/Resources/Images/Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/InductionMotors/torque_euqation.png\"/> </p>
+Where, <br>
+<<i>n</i> = Number of phases in connected power system <br>
+<i>R<sub>1</i></sub> = Stator resistance [ohm]<br>
```
correct it to `<i>R<sub>1</sub></i>`

mingzhe37 replied:
Thank you for viewing this, I have corrected such issues in all files.

JayHuLBL commented on Buildings/Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/InductionMotors/BaseClasses/MotorMachineInterface.mo:
```diff
@@ -0,0 +1,106 @@
+</html>",
+revisions="<html>
+<ul>
+<li>6 March 2019, by Yangyang Fu:<br>First implementation.</li>
```
check the format of writing the release note.

mingzhe37 replied:
Thank you for viewing, I've corrected the formatting of the release notes and other sections to make sure they match the other models in MBL.

JayHuLBL commented on Buildings/Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/InductionMotors/BaseClasses/MotorMachineInterface.mo:
```diff
@@ -0,0 +1,106 @@
+        rotation=0,
+        origin={-120,40})));
+  Modelica.Blocks.Interfaces.RealInput f(final quantity="Frequency", final unit="Hz")
+    "Controllale freuqency to the motor"
```
typo `Controllale`.

mingzhe37 replied:
Thank you for catching this, I have corrected the typo.

JayHuLBL commented on Buildings/Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/Coupled/Chiller.mo:
```diff
@@ -0,0 +1,284 @@
+
+  //Chiller parameters
+  parameter Modelica.Units.SI.HeatFlowRate QEva_flow_nominal(max=0) = -P_nominal * COP_nominal
+    "Nominal cooling heat flow rate (QEva_flow_nominal < 0)"
```
in comment, best would be not showing the parameter/variable name. So the comment should be `Nominal cooling heat flow rate (negative)`. Also, add
```
initial equation 
  assert(QEva_flow_nominal < 0, "Parameter QEva_flow_nominal must be negative.");
```

mingzhe37 replied:
Thank you for your suggestions! Originally I just referred to the chiller model in MBL (https://github.com/lbl-srg/modelica-buildings/blob/8ab33d18a40dec150af2d10335f31d7cd002db12/Buildings/Fluid/Chillers/BaseClasses/Carnot.mo#L11), It is definitely more helpful to understand with a clear description. I have revised the comment and added initial equation.

JayHuLBL commented on Buildings/Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/Coupled/Chiller.mo:
```diff
@@ -0,0 +1,284 @@
+  Modelica.Blocks.Interfaces.RealInput meaPoi "Measured value of control target"
+    annotation (Placement(transformation(extent={{-120,20},{-100,40}}),
+        iconTransformation(extent={{-120,20},{-100,40}})));
+  Modelica.Blocks.Interfaces.RealOutput P(final quantity="Power",final unit="W")
```
improve the text format

mingzhe37 replied:
Thank you for viewing, I've corrected the text format of all files to make sure they match the other models in MBL.

JayHuLBL commented on Buildings/Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/Coupled/Chiller.mo:
```diff
@@ -0,0 +1,284 @@
+        Line(points={{80,-30},{80,0}},                color={0,0,255})}),
+        defaultComponentName="chi",
+    Documentation(info="<html>
+<p>This is a model of a squirrel cage induction motor coupled chiller with ideal speed control. The model has electrical interfaces and can be used for simulating microgrids and discussing grid interactions.</p>
```
change the line. The text line is too long.

mingzhe37 replied:
Thank you for viewing, I have corrected the code length in each line, but I keep this issue in **script command** and **hyperlink in documents** unchanged. 
The reason is that I found the full path is used in the hyperlink in documents, it will invalidate the hyperlink if I break the full path into two lines, for example https://github.com/lbl-srg/modelica-buildings/pull/2955/files#diff-b87f3dc2806da6b599072686a2271acfef4b137dfb45f2adb1a164b05eb48409R157.
Does this sound reasonable to you? Or am I just not taking the correct approach to solve this problem?

JayHuLBL replied:
@mingzhe37 Yes, the hyperlink and script command should not be broken to separated lines.

JayHuLBL commented on Buildings/Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/Coupled/Chiller.mo:
```diff
@@ -0,0 +1,284 @@
+</html>",
+      revisions="<html>
+<ul>
+<li>October 15, 2021,
```
see the revision note format.

mingzhe37 replied:
Thank you for viewing, I've corrected the formatting of the release notes and other sections to make sure they match the other models in MBL.

JayHuLBL commented on Buildings/Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/Coupled/Examples/package.mo:
```diff
@@ -0,0 +1,9 @@
+
+
+annotation (Documentation(info="<html>
+<p>This package contains examples for the use of models that can be found in <a href=\"modelica://MotorDrive.Coupled\">MotorDrive.Coupled</a>.</p>
```
better use the full path link

mingzhe37 replied:
Thank you for catching this, I have corrected the path to the full path in all relevant files.

JayHuLBL commented on Buildings/Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/Coupled/Pump.mo:
```diff
@@ -0,0 +1,166 @@
+    redeclare replaceable Interfaces.Terminal_n terminal);
+
+  parameter Boolean addPowerToMedium=true
+    "Set to false to avoid any power (=heat and flow work) being added to medium (may give simpler equations)";
```
make the comment more clearer.

mingzhe37 replied:
Thank you for your suggestion! I have revised the comments in related models. Same as mentioned before, originally I just referred to the pump model in MBL (https://github.com/lbl-srg/modelica-buildings/blob/8ab33d18a40dec150af2d10335f31d7cd002db12/Buildings/Fluid/Movers/BaseClasses/PartialFlowMachine.mo#L41). However as this is an extension of original model, it would be better to use clear comment to improve readability of models, I have revised those comments accordingly.

JayHuLBL commented on Buildings/Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/InductionMotors/BaseClasses/package.mo:
```diff
@@ -0,0 +1,11 @@
+package BaseClasses "Package with base classes for MotorDrive.InductionMotors"
+  extends Modelica.Icons.BasesPackage;
+
+
```
remove empty lines.

mingzhe37 replied:
Thank you for your review, I have removed the unnecessary empty lines in all files.

JayHuLBL commented on Buildings/Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/ThermoFluid/package.mo:
```diff
@@ -0,0 +1,19 @@
+package ThermoFluid "Package with HVAC equipment models"
+    extends Modelica.Icons.Package;
+
+
```
empty lines.

mingzhe37 replied:
Thank you for reviewing these models, I have removed the unnecessary empty lines in all related files.

JayHuLBL reviewed (changes requested):
@mingzhe37 Please see inline comments. Would you please also improve the text formatting? I would like to do the second round review after the comments being addressed.

mingzhe37 commented:
> @mingzhe37 Please see inline comments. Would you please also improve the text formatting? I would like to do the second round review after the comments being addressed.

Sure Dr. Hu, thanks for your review, I will address these comments.

JayHuLBL commented on Buildings/Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/InductionMotors/BaseClasses/MotorMachineInterface.mo:
```diff
@@ -0,0 +1,130 @@
+The per-phase equivalent circuit model for an induction motor supplied by 
+balanced three-phase supply based on the transformer model is represented below. 
+</p>
+<p align=\"center\"><img src=\"modelica://Buildings/Resources/Images/Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/InductionMotors/induction_motor_equivalent_circuit.png\"/> 
```
Check the syntax, it will cause html check warning as: `<img> lacks "alt" attribute`.
see for example here:
https://github.com/lbl-srg/modelica-buildings/blob/79f20f28c229fca01a802b61a7b391bbf585109e/Buildings/Controls/OBC/CDL/Continuous/Acos.mo#L55

JayHuLBL commented on Buildings/Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/InductionMotors/BaseClasses/MotorMachineInterface.mo:
```diff
@@ -0,0 +1,130 @@
+<p>
+The electromagnetic torque generated by motor is calculated as follows. 
+</p>
+<p align=\"center\"><img src=\"modelica://Buildings/Resources/Images/Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/InductionMotors/torque_euqation.png\"/> 
```
`<img> lacks "alt" attribute`.

JayHuLBL commented on Buildings/Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/ThermoFluid/Chiller.mo:
```diff
@@ -0,0 +1,314 @@
+efficient buildings.&quot;&nbsp;<i>IEEE Transactions on Industrial Informatics</i>&nbsp;15.8 (2019): 4384-4393.</a></span>
+</p>
+<p>
+<span style=\"font-family: Arial;\"><a href=\"https://www.proquest.com/docview/2414053191?pq-origsite=gscholar&fromopenview=true\">
```
You may delete `&fromopenview=true`. This will cause html check error:
```
line 67 column 110 - Warning: unescaped & or unknown entity "&fromopenview"
```
The hyperlink will still be valid.

JayHuLBL commented on Buildings/Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/ThermoFluid/HeatPump.mo:
```diff
@@ -0,0 +1,311 @@
+efficient buildings.&quot;&nbsp;<i>IEEE Transactions on Industrial Informatics</i>&nbsp;15.8 (2019): 4384-4393.</a></span>
+</p>
+<p>
+<span style=\"font-family: Arial;\"><a href=\"https://www.proquest.com/docview/2414053191?pq-origsite=gscholar&fromopenview=true\">
```
delete `&fromopenview=true`

JayHuLBL commented on Buildings/Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/InductionMotors/BaseClasses/MotorMachineInterface.mo:
```diff
@@ -0,0 +1,130 @@
+          fillPattern=FillPattern.Solid),
+          Text(
+          extent={{-82,162},{82,116}},
+          lineColor={0,0,255},
```
should be `textColor{0,0,255}`, rather than `lineColor{}`.

JayHuLBL commented on Buildings/Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/InductionMotors/SquirrelCageDrive.mo:
```diff
@@ -0,0 +1,226 @@
+within Buildings.Electrical.AC.ThreePhasesBalanced.Loads.MotorDrive.InductionMotors;
+model SquirrelCageDrive "Squirrel cage type induction motor with electrical interface and built-in speed control"
+  extends Buildings.Electrical.Interfaces.PartialOnePort(
+  redeclare package PhaseSystem = Buildings.Electrical.PhaseSystems.OnePhase,
```
one tab inside.

JayHuLBL commented on Buildings/Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/InductionMotors/SquirrelCageDrive.mo:
```diff
@@ -0,0 +1,226 @@
+  Modelica.Mechanics.Rotational.Interfaces.Flange_b shaft "Mechanical connector"
+    annotation (Placement(transformation(extent={{90,-10},{110,10}})));
+  Buildings.Electrical.AC.ThreePhasesBalanced.Loads.MotorDrive.InductionMotors.BaseClasses.MotorMachineInterface torSpe(
+  final n=n,
```
one tab inside.

JayHuLBL commented on Buildings/Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/ThermoFluid/Pump.mo:
```diff
@@ -0,0 +1,158 @@
+model Pump
+  "Fan or pump with mechanical interface"
+  extends Buildings.Fluid.Interfaces.PartialTwoPort(
+  port_a(p(start=Medium.p_default)),
```
one tab inside.

JayHuLBL reviewed (changes requested):
@mingzhe37 Please see inline comments.

mingzhe37 commented:
Hi Dr. Hu @JayHuLBL, thank you for your review, I addressed these comments and applying to all related models. Please let me know if any revisions are needed.
